api/tools/src/main/java/org/pmiops/workbench/tools/ExportWorkspaceData.java

```diff
@@ -190,15 +142,79 @@ public CommandLineRunner run(
+          new FileWriter(opts.getOptionValue(exportFilenameOpt.getLongOpt()))) {
+        StatefulBeanToCsv beanWriter =
+            new StatefulBeanToCsvBuilder(writer).withMappingStrategy(mappingStrategy).build();
 
-      beanWriter.write(rows);
-
-      writer.close();
+        beanWriter.write(rows);
+      }
     };
   }
 
+  private WorkspaceExportRow toWorkspaceExportRow(DbWorkspace workspace) {
+    WorkspaceExportRow row = new WorkspaceExportRow();
+    row.setCreatorContactEmail(workspace.getCreator().getContactEmail());
+    row.setCreatorUsername(workspace.getCreator().getEmail());
+    row.setCreatorFirstSignIn(dateFormat.format(workspace.getCreator().getFirstSignInTime()));
+    row.setCreatorRegistrationState(workspace.getCreator().getDataAccessLevelEnum().toString());
+
+    row.setProjectId(workspace.getWorkspaceNamespace());
+    row.setName(workspace.getName());
+    row.setCreatedDate(dateFormat.format(workspace.getCreationTime()));
+
+    try {
+      row.setCollaborators(
+          extractAclResponse(
+                  workspacesApi.getWorkspaceAcl(
+                      workspace.getWorkspaceNamespace(), workspace.getFirecloudName()))
+              .keySet().stream()
+              .collect(Collectors.joining(",\n")));
+    } catch (ApiException e) {
+      row.setCollaborators("Error: Not Found");
+    }
+
+    Collection<DbCohort> cohorts = cohortDao.findByWorkspaceId(workspace.getWorkspaceId());
+    row.setCohortNames(
+        cohorts.stream().map(cohort -> cohort.getName()).collect(Collectors.joining(",\n")));
+    row.setCohortCount(String.valueOf(cohorts.size()));
+
+    Collection<DbConceptSet> conceptSets =
+        conceptSetDao.findByWorkspaceId(workspace.getWorkspaceId());
+    row.setConceptSetNames(
+        conceptSets.stream()
+            .map(conceptSet -> conceptSet.getName())
+            .collect(Collectors.joining(",\n")));
+    row.setConceptSetCount(String.valueOf(conceptSets.size()));
+
+    Collection<DbDataset> datasets = dataSetDao.findByWorkspaceId(workspace.getWorkspaceId());
+    row.setDatasetNames(
+        datasets.stream().map(dataSet -> dataSet.getName()).collect(Collectors.joining(",\n")));
+    row.setDatasetCount(String.valueOf(datasets.size()));
+
+    try {
+      Collection<FileDetail> notebooks =
+          notebooksService.getNotebooks(
+              workspace.getWorkspaceNamespace(), workspace.getFirecloudName());
+      row.setNotebookNames(
+          notebooks.stream()
+              .map(notebook -> notebook.getName())
+              .collect(Collectors.joining(",\n")));
+      row.setNotebooksCount(String.valueOf(notebooks.size()));
+    } catch (NotFoundException e) {
+      row.setNotebookNames("Error: Not Found");
+      row.setNotebooksCount("N/A");
+    }
+
+    DbWorkspaceFreeTierUsage usage = workspaceFreeTierUsageDao.findOneByWorkspace(workspace);
+    row.setWorkspaceSpending(usage == null ? "0" : String.valueOf(usage.getCost()));
```

Comment:
Please document (and present with the spreadsheet) any special cases like this, where we are promoting missing values to defaults. It can make a difference for analytics. For example, when computing averages, null values should be left out of the set, whereas zeros are typically valid data points.